GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3015,37 +3015,46 @@ public void ChangeTerminalActiveFolder(string path)
                 return;
             }
 
-            if (AppSettings.ConEmuTerminal.ValueOrDefault == "bash")
+            switch (AppSettings.ConEmuTerminal.ValueOrDefault)
             {
-                if (PathUtil.TryConvertWindowsPathToPosix(path, out var posixPath))
-                {
-                    ClearTerminalCommandLineAndRunCommand("cd " + posixPath);
-                }
-            }
-            else if (AppSettings.ConEmuTerminal.ValueOrDefault == "powershell")
-            {
-                ClearTerminalCommandLineAndRunCommand("cd \"" + path + "\"");
-            }
-            else
-            {
-                ClearTerminalCommandLineAndRunCommand("cd /D \"" + path + "\"");
+                case "bash":
+                    if (PathUtil.TryConvertWindowsPathToPosix(path, out var posixPath))
+                    {
+                        ClearTerminalCommandLineWithMacroAndRunCommand($"cd {posixPath.QuoteNE()}");
+                    }
+
+                    break;
+                case "cmd":
+                    ClearTerminalCommandLineWithEscapeAndRunCommand($"cd /D {path.QuoteNE()}");
+                    break;
+                case "powershell":
```

Comment:
I think I would like to be the default behavior. That way, there is a probability that work out of the box if we add another shell.

What do you think about it?

Reply:
I don't have a strong opinion here. I did open an issue to request custom shells (or at least add `pwsh`), but I suspect that will require other changes in this code that can't really be planned for at this point. I can see your case for providing a default, but also think there's a case for no-op, since right now we are only expecting three values. So I could go either way.

Reply:
I'm happy with the current proposal.